src/serialize/value.rs

```diff
@@ -0,0 +1,1050 @@
+                typ,
+                BuiltinTypeCheckErrorKind::MismatchedType { expected },
+            )));
+        };
+
+        value.serialize(typ, cell_writer)
+    }
+
+    fn serialize_arbitrary_value<'b>(
+        &self,
+        typ: &ColumnType,
+        cell_writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        match typ {
+            ColumnType::Native(native_type) => {
+                self.serialize_natives(typ, native_type, cell_writer)
+            }
+
+            ColumnType::Collection {
+                typ: collection_typ,
+                ..
+            } => match collection_typ {
+                CollectionType::List(_) => {
+                    let Ok(list) = PyListWrapper::new(self) else {
+                        let expected = self.map_type_to_expected()?;
+                        return Err(SerializationError::new(mk_typck_err::<PyList>(
+                            typ,
+                            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+                        )));
+                    };
+
+                    list.serialize(typ, cell_writer)
+                }
+
+                CollectionType::Map(_, _) => {
+                    let Ok(map) = PyMapWrapper::new(self) else {
+                        let expected = self.map_type_to_expected()?;
+                        return Err(SerializationError::new(mk_typck_err::<PyMapping>(
+                            typ,
+                            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+                        )));
+                    };
+
+                    map.serialize(typ, cell_writer)
+                }
+
+                CollectionType::Set(_) => {
+                    let Ok(set) = PySetWrapper::new(self) else {
+                        let expected = self.map_type_to_expected()?;
+                        return Err(SerializationError::new(mk_typck_err::<PySet>(
+                            typ,
+                            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+                        )));
+                    };
+
+                    set.serialize(typ, cell_writer)
+                }
+
+                _ => {
+                    let python_collection_name = self.get_type().name().map_err(|e| {
+                        SerializationError::new(PythonDriverSerializationError::PythonError(e))
+                    })?;
+
+                    let python_collection_name =
+                        python_collection_name.extract::<&str>().map_err(|e| {
+                            SerializationError::new(PythonDriverSerializationError::PythonError(e))
+                        })?;
+
+                    Err(SerializationError::new(
+                        PythonDriverSerializationError::UnknownCollectionType(
+                            python_collection_name.to_string(),
+                        ),
+                    ))
+                }
```

Comment:
:wrench: The same problem: you're reporting Python type name as unknown, while the problem is with the CQL CollectionType.